### PR TITLE
LD-478 Move favorites sync up to app level

### PR DESCRIPTION
**Pull request by rjspencer**

### CODE REVIEW

File: src/services/pmo/favorite/FavoritesSync.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  return
+}
+
+const FavoritesSync: FC = () => {
+  useFavoritesSync()
+
+  return null
+}
+
+export default FavoritesSync
```
Comment (chasenlehara) [inline src/services/pmo/favorite/FavoritesSync.tsx:29]: Out of curiosity, why is this a component and not only the `useFavoritesSync()` Hook used in App?

File: src/services/pmo/favorite/FavoritesSync.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const { isConnected } = useNetInfo()
+  const { syncWithServer, localFavorites } = useFavorites(user?.id)
+
+  useEffect(() => {
+    if (user && isConnected && localFavorites) {
+      syncWithServer()
+    }
+  }, [isConnected, localFavorites, syncWithServer, user])
+
+  return
```
Comment (chasenlehara) [inline src/services/pmo/favorite/FavoritesSync.tsx:20]: This also sets off a little bell in my mind… it makes me think that “favorites” should be in context (so there’s only one instance) and `localFavorites` should be provided by the context.

File: src/services/pmo/favorite/FavoritesSync.tsx (hunk 1 above)
Comment (rjspencer) [reply src/services/pmo/favorite/FavoritesSync.tsx:29]: Just makes the App file cleaner. The hook needs to be inside the AuthProvider. To just use the hook I would need to place the hook inside the navigation component.

Comment (christopherjbaker) [reply src/services/pmo/favorite/FavoritesSync.tsx:29]: I was going to ask the same. These kinds of "headless components" (as I call them) aren't common, but they do make things cleaner, and sometimes simplify things a lot when context is in play.

File: src/services/pmo/favorite/FavoritesSync.tsx (hunk 2 above)
Comment (christopherjbaker) [reply src/services/pmo/favorite/FavoritesSync.tsx:20]: The way its used, there will only ever be one. That breaks the reusability of the component but also makes it a whole lot simpler to write and use, so it's a tradeoff; neither option is ideal. With how simple it is, I'd leave it like this.

**Review by christopherjbaker — APPROVED** [review]

**Pull request merged**